File: frontend/src/types/producto.ts
// Producto type definitions based on backend API schemas

export interface ProductoBase {
  nombre: string;
  descripcion?: string | null;
  precio: number; // Converted from Decimal
  tipo_iva: number; // Converted from Decimal, default: 21.00
  es_servicio: boolean;
  codigo?: string | null;
  activo: boolean;
}

export interface ProductoCreate extends ProductoBase {
  // All fields from ProductoBase, with required fields enforced
  tipo_iva: number; // Default to 21.00
  es_servicio: boolean; // Default to false
  activo: boolean; // Default to true
}

export interface ProductoUpdate {
  nombre?: string;
  descripcion?: string | null;
  precio?: number;
  tipo_iva?: number;
  es_servicio?: boolean;
  codigo?: string | null;
  activo?: boolean;
}

export interface Producto extends ProductoBase {
  id: number;
  user_id: string;
  created_at: string; // ISO 8601 datetime string
  updated_at: string; // ISO 8601 datetime string
}

export type ProductosListResponse = Producto[];
export type ProductoResponse = Producto;

File: backend/app/schemas/producto.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime
from typing import Optional
from decimal import Decimal

class ProductoBase(BaseModel):
    nombre: str
    descripcion: Optional[str] = None
    precio: Decimal
    tipo_iva: Decimal = Decimal("21.00")
    es_servicio: bool = False
    codigo: Optional[str] = None
    activo: bool = True

class ProductoCreate(ProductoBase):
    pass

class ProductoUpdate(BaseModel):
    nombre: Optional[str] = None
    descripcion: Optional[str] = None
    precio: Optional[Decimal] = None
    tipo_iva: Optional[Decimal] = None
    es_servicio: Optional[bool] = None
    codigo: Optional[str] = None
    activo: Optional[bool] = None

class ProductoResponse(ProductoBase):
    id: int
    user_id: str
    created_at: datetime
    updated_at: datetime
    
    model_config = ConfigDict(from_attributes=True)